server/service/src/invoice_line/stock_out_line/update/validate.rs

```diff
@@ -63,6 +65,46 @@ pub fn validate(
     check_location_on_hold(&batch_pair.main_batch).map_err(|e| match e {
         LocationIsOnHoldError::LocationIsOnHold => LocationIsOnHold,
     })?;
+
+    // If we have an allocated_date older than 24hours, we need to calculate the historical stock line to see if we would have enough stock at that time
+    let batch_pair = if let Some(allocated_date) = invoice.allocated_datetime.clone() {
+        if allocated_date < chrono::Utc::now().naive_utc() - chrono::Duration::hours(24) {
+            let historical_stock_lines =
+                get_historical_stock_lines(ctx, store_id, &item.id, &allocated_date).map_err(
```

Comment:
I agree, it depends "where" it's easy to use, but yes, that was my point about refactoring it.
The idea was this can replace the existing frontend call for stocklines, but need to balance that with internal uses like this...